conda-store-server/conda_store_server/action/generate_lockfile.py

```diff
@@ -37,6 +37,16 @@ def print_cmd(cmd):
     print_cmd(["conda", "config", "--show"])
     print_cmd(["conda", "config", "--show-sources"])
 
+    # conda-lock ignores variables defined in the specification, so this code
+    # gets the value of CONDA_OVERRIDE_CUDA and passes it to conda-lock via
+    # the with_cuda parameter, see:
+    # https://github.com/conda-incubator/conda-store/issues/719
+    # https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html#overriding-detected-packages
+    if specification.variables is not None:
+        cuda_version = specification.variables.get("CONDA_OVERRIDE_CUDA")
```

Comment:
The `CONDA_OVERRIDE_CUDA` variable could come in as a string (expected by conda-lock) or a float. 

The example from the description of this PR
```yaml
...
variables:
  CONDA_OVERRIDE_CUDA: '12.0'
```

Will come through as a string and be forwarded to conda-lock, but the example in the original issue has:

```yaml
...
variables:
  CONDA_OVERRIDE_CUDA: 11.8
```

Which would be parsed as float and be forwarded to conda-lock as a float. 

I don't see a test update associated with this, so I am assuming we don't have any coverage here, could you try this out locally to see if we need to convert the value as we pull it off the specification?

Reply:
Added a print to `action_solve_lockfile` to verify. It's always a string, even if you don't quote it. Tried with 12, 12.0, and 12.2.

(In fact, the admin UI renders previously unquoted versions in quotes once you submit an env.)

So no action is needed here.

----

You also asked offline whether using `CONDA_OVERRIDE_CUDA` on a machine without a GPU allows for using PyTorch with CUDA on a GPU-enabled machine later. I've tried the following:

- On a machine with an Intel GPU (so no CUDA):
  - added the env from the top comment to conda-store, which has `CONDA_OVERRIDE_CUDA: '12.0'`
  - waited for it to generate the lockfile (but didn't wait for it to build)
  - downloaded the lockfile
- On a machine with an NVIDIA GPU (and CUDA configured):
  - created an env from the downloaded lockfile via `conda-lock install -n mytest1 --micromamba ~/mytest1.json`
  - activated the env
  - confirmed that CUDA works:
  
```
>>> import torch
>>> torch.__file__
'/home/nkaretnikov/.conda/envs/mytest1/lib/python3.12/site-packages/torch/__init__.py'
>>> print(torch.version.cuda)
12.0
>>> x = torch.tensor(1, device='cuda')
>>> x.device
device(type='cuda', index=0)
>>> x
tensor(1, device='cuda:0')
```

Please let me know if you have any other questions!